server/planning/assignments/assignments_unlink.py

```diff
@@ -28,55 +28,68 @@ def create(self, docs):
         ids = []
         production = get_resource_service('archive')
         assignments_service = get_resource_service('assignments')
-        items = []
+        updated_items = []
+        actioned_item = {}
 
         for doc in docs:
             assignment = assignments_service.find_one(req=None, _id=doc.pop('assignment_id'))
             assignments_service.validate_assignment_action(assignment)
-            item = production.find_one(req=None, _id=doc.pop('item_id'))
-            # Boolean set to true if the unlink is as the result of spiking the content item
-            spike = doc.pop('spike', False)
-
-            # Set the state to 'assigned' if the item is 'submitted'
-            updates = {'assigned_to': deepcopy(assignment.get('assigned_to'))}
-            updates['assigned_to']['state'] = ASSIGNMENT_WORKFLOW_STATE.ASSIGNED
-            assignments_service.patch(assignment[config.ID_FIELD], updates)
-
-            production.system_update(
-                item[config.ID_FIELD],
-                {'assignment_id': None},
-                item
-            )
-
-            # Update published collection
-            if item.get(ITEM_STATE) in PUBLISH_STATES:
-                get_resource_service('published').update_published_items(
+            actioned_item_id = doc.pop('item_id')
+            actioned_item = production.find_one(req=None, _id=actioned_item_id)
+
+            # To support 'unlink as update' we need to link only that update. Not the entire chain
+            if actioned_item and actioned_item.get('rewrite_of'):
+                items = production.find({
+                    'family_id': actioned_item.get('family_id'),
```

Comment:
You will have to use the `event_id` here. 
See lines for code:
https://github.com/superdesk/superdesk-core/blob/master/apps/archive/archive_rewrite.py#L107
https://github.com/superdesk/superdesk-core/blob/master/apps/archive/archive_rewrite.py#L153
https://github.com/superdesk/superdesk-core/blob/master/apps/archive/archive_rewrite.py#L292
https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/archive/services/FamilyService.ts#L98

Original logic was based on find_one i.e. find by `_id` but now it different field. We have to use `elastic index` over here for `event_id` for archive and published endpoints. If the item is corrected/killed you will get more docs for single item_id.  There is also `rewrite_sequence` field not sure if you need it to order deliveries.